src/main/java/org/sonarlint/intellij/inspection/SonarLintGlobalInspection.java

```diff
@@ -141,20 +144,68 @@ public void visitFile(PsiFile psiFile) {
   }
 
   ProblemDescriptor getProblemDescriptor(Issue issue, PsiFile psiFile, @NotNull InspectionManager manager, boolean isLocal) {
-    if (issue.getLine() != null) {
-      return manager.createProblemDescriptor(psiFile,
-        InspectionUtils.getLineRange(psiFile, issue.getLine()),
+    Integer startLine = issue.getStartLine();
+    if (startLine == null) {
+      // No line, global issue
+      return createProblemDescriptorFileLevel(issue, psiFile, manager, isLocal);
+    }
+    Integer startLineOffset = issue.getStartLineOffset();
+    if (startLineOffset == null) {
+      // No offset, either old SQ version or language plugin doesn't provide data. Fallback to text range
+      return createProblemDescriptorPreciseTextRange(issue, psiFile, manager, isLocal, InspectionUtils.getLineRange(psiFile, startLine));
+    }
+    Integer endLine = issue.getEndLine();
+    Integer endLineOffset = issue.getEndLineOffset();
+    int ijStartLine = startLine - 1;
+    int ijEndLine = endLine - 1;
+    final Document document = PsiDocumentManager.getInstance(psiFile.getProject()).getDocument(psiFile);
+    PsiElement startElement;
+    PsiElement endElement;
+    if (document == null) {
+      // I'm not sure when this can occurs. Create a global issue
+      return createProblemDescriptorFileLevel(issue, psiFile, manager, isLocal);
+    }
+    return tryCreateProblemDescriptorPrecisePsiElement(issue, psiFile, manager, isLocal, startLineOffset, endLineOffset, ijStartLine, ijEndLine, document);
+  }
+
+  @NotNull
+  private ProblemDescriptor tryCreateProblemDescriptorPrecisePsiElement(Issue issue, PsiFile psiFile, @NotNull InspectionManager manager, boolean isLocal, Integer startLineOffset, Integer endLineOffset, int ijStartLine, int ijEndLine, Document document) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 255 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00103)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 9 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00107)